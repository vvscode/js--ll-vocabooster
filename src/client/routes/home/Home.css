@import '@client/styles/variables.css';

.layout {
  background: var(--light-background);
  width: 100%;
  padding-bottom: 50px;
  position: relative;
  height: calc(100vh - 150px);
}
