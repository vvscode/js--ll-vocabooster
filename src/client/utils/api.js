const getFormdata = data => {
  const formData = new FormData();
  Object.keys(data).forEach(key => formData.append(key, data[key]));
  return formData;
};

export const getWords = text =>
  fetch('/api/vocagrabber', {
    method: 'POST',
    body: getFormdata({ text }),
  })
    .then(r => r.json())
    .then(({ data: list }) =>
      list.map((item, index) => ({ ...item, id: index })),
    );

export const addWordsToLingualeoVocabulary = words => Promise.resolve();
