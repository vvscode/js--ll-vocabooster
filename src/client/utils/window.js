export const { document } = window;

export const setTitle = title => {
  window.document.title = title;
};

export default window;
