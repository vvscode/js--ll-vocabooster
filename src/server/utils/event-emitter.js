import { EventEmitter } from 'events';

const customEventEmitter = new EventEmitter();

export default customEventEmitter;
