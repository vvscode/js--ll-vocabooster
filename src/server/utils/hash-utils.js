import hash from 'object-hash';

const hashAny = (value, keysToExclude = []) => {
  if (!value) return {};

  return {
    val: value,
    key: hash(value, {
      algorithm: 'md5',
      excludeKeys: key => keysToExclude.includes(key),
    }),
  };
};

const hashArray = arr => {
  const newArray = [];

  arr.forEach(item => {
    newArray.push(hashAny(item));
  });

  return newArray;
};

export { hashArray, hashAny };
