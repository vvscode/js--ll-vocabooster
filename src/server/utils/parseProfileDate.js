import dateformat from 'dateformat';

export default function parseProfileDate(dateString) {
  try {
    return dateformat(dateString, 'isoDate');
  } catch (error) {
    console.error(error);
    return null;
  }
}
