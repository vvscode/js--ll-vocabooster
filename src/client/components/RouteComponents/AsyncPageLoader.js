import Loadable from 'react-loadable';
import { setTitle } from 'client/utils/window';

import Loading from './Loading';

const DEFAULT_TITLE = 'Lingualeo VocaBooster';
const TITLE_PREFIX = '[Lingualeo VocaBooster]: ';
const TITLE_POSTFIX = ' *';

const PageComponent = path =>
  // https://github.com/jamiebuilds/react-loadable
  Loadable({
    loader: (...args) =>
      import(`client/routes/${path}`)
        // it also handles async `action` hooks
        .then(({ default: action }) => action(...args))
        .then(({ component, title }) => {
          let newTitle;
          if (!title) {
            newTitle = DEFAULT_TITLE;
          } else {
            newTitle = `${TITLE_PREFIX}${title}${TITLE_POSTFIX}`;
          }
          setTitle(newTitle);
          return component;
        }),
    loading: Loading,
  });

export default PageComponent;
