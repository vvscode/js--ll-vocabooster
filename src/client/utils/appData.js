import window from './window';

const data = (window && window.App) || {};

export const isLoggedIn = () =>
  !!data.user && Object.keys(data.user).length > 0;

export default data;
