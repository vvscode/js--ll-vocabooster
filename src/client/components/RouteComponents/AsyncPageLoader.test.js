/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import { mount } from 'enzyme';
import Home from 'client/routes/home';
import { setTitle } from 'client/utils/window';
import AsyncPageLoader from './AsyncPageLoader';

jest.mock('client/routes/home');
jest.mock('client/utils/window');
// jest.useFakeTimers();

const waitFor = delay => new Promise(resolve => setTimeout(resolve, delay));
const DummyComponent = () => <div>Dummy</div>;

describe('AsyncPageLoader', () => {
  beforeEach(() => {
    Home.mockReturnValue({
      component: () => <DummyComponent />,
    });
  });

  describe('render', () => {
    test('renders correctly', async () => {
      const AsyncComponent = AsyncPageLoader('home');
      const component = mount(<AsyncComponent />);
      expect(component).toMatchSnapshot('loading');
      await waitFor(100);
      component.update();
      expect(component).toMatchSnapshot('loaded');
      expect(setTitle).toBeCalledWith('Lingualeo VocaBooster');
    });

    test('renders correctly from async module', async () => {
      Home.mockReturnValue(
        Promise.resolve({
          title: 'Some title',
          component: () => <DummyComponent />,
        }),
      );
      const AsyncComponent = AsyncPageLoader('home');
      const component = mount(<AsyncComponent />);
      expect(component).toMatchSnapshot('loading');
      await waitFor(100);
      component.update();
      expect(component).toMatchSnapshot('loaded');
    });
  });

  describe('handle title', () => {
    test('set default title if not provided', async () => {
      const AsyncComponent = AsyncPageLoader('home');
      mount(<AsyncComponent />);
      await waitFor(100);
      expect(setTitle).toBeCalledWith('Lingualeo VocaBooster');
    });

    test('set custom title with prefix/postfix if provided', async () => {
      Home.mockReturnValue({
        title: 'Some title',
        component: () => <DummyComponent />,
      });
      const AsyncComponent = AsyncPageLoader('home');
      mount(<AsyncComponent />);
      await waitFor(100);
      expect(setTitle).toBeCalledWith('[Lingualeo VocaBooster]: Some title *');
    });
  });
});
