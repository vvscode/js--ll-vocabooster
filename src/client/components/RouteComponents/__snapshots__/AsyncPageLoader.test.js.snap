// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsyncPageLoader render renders correctly from async module: loaded 1`] = `
<LoadableComponent>
  <component>
    <DummyComponent>
      <div>
        Dummy
      </div>
    </DummyComponent>
  </component>
</LoadableComponent>
`;

exports[`AsyncPageLoader render renders correctly from async module: loading 1`] = `
<LoadableComponent>
  <_default
    error={null}
    isLoading={true}
    pastDelay={false}
    timedOut={false}
  >
    <div
      className="loading"
    >
      Loading...
    </div>
  </_default>
</LoadableComponent>
`;

exports[`AsyncPageLoader render renders correctly: loaded 1`] = `
<LoadableComponent>
  <component>
    <DummyComponent>
      <div>
        Dummy
      </div>
    </DummyComponent>
  </component>
</LoadableComponent>
`;

exports[`AsyncPageLoader render renders correctly: loading 1`] = `
<LoadableComponent>
  <_default
    error={null}
    isLoading={true}
    pastDelay={false}
    timedOut={false}
  >
    <div
      className="loading"
    >
      Loading...
    </div>
  </_default>
</LoadableComponent>
`;
